### PR TITLE
[PL-38452] Add build custom images using delegate Dockerfile topic

### PR DESCRIPTION
**Pull request by brian-f-harness**
Add build custom images using delegate Dockerfile topic

For reviewers, [a preview](https://6463b515634eb900a27b98b5--harness-developer.netlify.app/docs/platform/Delegates/manage-delegates/build-custom-images-delegate-dockerfile) is available.

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — DISMISSED** [review]

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: Build custom images using Dockerfile
+description: This topic describes how to build custom images using the Harness Delegate Dockerfile.
```
Comment (michaelcretzman) [inline docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:3]:
```suggestion
title: Build custom delegate images using Dockerfile
description: This topic describes how to build custom delegate images using the Harness Delegate Dockerfile.
```

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+You can use the Harness Delegate Dockerfile to build custom delegate images. The [Dockerfile](https://docs.docker.com/engine/reference/builder/) is available in the [delegate Dockerfile repository](https://github.com/harness/delegate-dockerfile). 
+
+The repository includes the `Dockerfile-minimal` and `Dockerfile-ubuntu` versions. 
+
+:::info note
+If you build and use custom images, you can choose to set automatic upgrades off or on for Kubernetes delegates. To learn more about automatic upgrades with custom images, go to [Use automatic upgrade with custom delegate images](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration#use-automatic-upgrade-with-custom-delegate-images).
+
+For more information on delegate automatic upgrades and expiration policy, go to [Delegate automatic upgrades and expiration policy](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration).
```
Comment (michaelcretzman) [inline docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:14]:
```suggestion
For more information on delegate automatic upgrades and the delegate expiration policy, go to [Delegate automatic upgrades and expiration policy](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration).
```

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+description: This topic describes how to build custom images using the Harness Delegate Dockerfile.
+sidebar_position: 7
+---
+
+You can use the Harness Delegate Dockerfile to build custom delegate images. The [Dockerfile](https://docs.docker.com/engine/reference/builder/) is available in the [delegate Dockerfile repository](https://github.com/harness/delegate-dockerfile). 
+
+The repository includes the `Dockerfile-minimal` and `Dockerfile-ubuntu` versions. 
+
+:::info note
+If you build and use custom images, you can choose to set automatic upgrades off or on for Kubernetes delegates. To learn more about automatic upgrades with custom images, go to [Use automatic upgrade with custom delegate images](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration#use-automatic-upgrade-with-custom-delegate-images).
```
Comment (michaelcretzman) [inline docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:12]:
```suggestion
If you build and use custom images, you can choose to enable or disable automatic upgrades for Kubernetes delegates. To learn more about automatic upgrades with custom images, go to [Use automatic upgrade with custom delegate images](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration#use-automatic-upgrade-with-custom-delegate-images).
```

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+For more information on delegate automatic upgrades and expiration policy, go to [Delegate automatic upgrades and expiration policy](https://developer.harness.io/docs/platform/Delegates/install-delegates/delegate-upgrades-and-expiration).
+:::
+
+## Dockerfile tools
+
+Use Dockerfile to create the delegate image with tools. This image includes the following tools by default.
```
Comment (michaelcretzman) [inline docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:19]:
```suggestion
You can include third party tools with your delegate when you use the delegate Dockerfile. The image includes the following tools by default.
```

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+* `kubectl` v1.24.3
```
Comment (michaelcretzman) [inline docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:21]: You're going to have to update these versions whenever our support changes. Consider linking to this list in one place. It should be in a repo. Ask your Eng for the link. Or, you can link to the list in the Supported platforms topic:

https://developer.harness.io/docs/getting-started/supported-platforms-and-technologies#sdks-installed-with-the-delegate

**Review by michaelcretzman — DISMISSED** [review]

File: docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md (hunk 5 above)
Comment (brian-f-harness) [reply docs/platform/2_Delegates/manage-delegates/build-custom-images-delegate-dockerfile.md:21]: @michaelcretzman Good point. I added a link here and will remove the list.

**Pull request merged**